### DISCUSSION TITLE
Hosting on Github Pages

### DISCUSSION THREAD

**Discussion by ayushsisodia7** [community/community#60572] [New to GitHub]
Hi Everyone,
I have got my domain(.me) after getting student developer pack.
I have created a basic site.
But how to host it on github pages?
Please help and guide me.

**Comment by nirav-gajera** [comment]
1. Create a new repository: Go to GitHub and create a new repository with the name `<your-username>.github.io`, where `<your-username>` is your GitHub username. This repository will serve as the hosting repository for your website.

2. Upload your website files: Clone the newly created repository to your local machine. Add your website files (HTML, CSS, JavaScript, etc.) to the repository and commit the changes. Make sure you have an index.html file in the root directory, as this will be the main page of your website.

3. Configure custom domain: In your repository settings, scroll down to the GitHub Pages section. Under the Custom domain field, enter your domain name (e.g., yourdomain.me). Save the changes.

4. Configure DNS settings: Access the DNS settings of your domain registrar (the service where you purchased your domain). Create a new DNS record of type CNAME with the host value as www and the value as `<your-username>.github.io.` This will point your domain to the GitHub Pages hosting.

5. Set up HTTPS (optional): GitHub Pages supports HTTPS encryption for custom domains. To enable HTTPS, select the "Enforce HTTPS" option in your repository settings under the GitHub Pages section.

6. Wait for propagation: DNS changes may take some time to propagate. It can take a few minutes to a few hours for your domain to start pointing to your GitHub Pages repository.

7. Once the DNS changes have propagated, you can access your website using your custom domain (yourdomain.me in this example). GitHub Pages will automatically build and serve your website from the repository.

**Reply by ayushsisodia7** [reply]
Thank you for the steps, will implement and revert back

**Comment by Victor-Njoroge** [comment]
1. You can create a new branch in your directory `git branch gh-pages`
2. Then git checkout to the gh-pages
3. Finally git push origin gh- pages

**Comment by Victor-Njoroge** [comment]
Also you can use apps such us vercel or netlify you can host your website there
